internal/interceptor/outbound.go

```diff
@@ -21,46 +21,173 @@
+	// DirectOnewayOutbound defines transport interceptor for `OnewayOutbound`s.
 	//
-	// OnewayOutbound interceptor MAY do zero or more of the following: change the
+	// DirectOnewayOutbound interceptor MAY do zero or more of the following: change the
 	// context, change the request, change the returned ack, handle the returned
 	// error, call the given outbound zero or more times.
 	//
-	// OnewayOutbound interceptor MUST always return an Ack (nil or not) or an
+	// DirectOnewayOutbound interceptor MUST always return an Ack (nil or not) or an
 	// error, and they MUST be thread-safe.
 	//
-	// OnewayOutbound interceptor is re-used across requests and MAY be called
+	// DirectOnewayOutbound interceptor is re-used across requests and MAY be called
 	// multiple times on the same request.
-	OnewayOutbound = middleware.OnewayOutbound
+	DirectOnewayOutbound interface {
+		CallOneway(ctx context.Context, request *transport.Request, out UnchainedOnewayOutbound) (transport.Ack, error)
+	}
 
-	// StreamOutbound defines transport interceptor for `StreamOutbound`s.
+	// DirectStreamOutbound defines transport interceptor for `StreamOutbound`s.
 	//
-	// StreamOutbound interceptor MAY do zero or more of the following: change the
+	// DirectStreamOutbound interceptor MAY do zero or more of the following: change the
 	// context, change the requestMeta, change the returned Stream, handle the
 	// returned error, call the given outbound zero or more times.
 	//
-	// StreamOutbound interceptor MUST always return a non-nil Stream or error,
-	// and they MUST be thread-safe
+	// DirectStreamOutbound interceptor MUST always return a non-nil Stream or error,
+	// and they MUST be thread-safe.
 	//
-	// StreamOutbound interceptors is re-used across requests and MAY be called
+	// DirectStreamOutbound interceptors is re-used across requests and MAY be called
 	// multiple times on the same request.
-	StreamOutbound = middleware.StreamOutbound
+	DirectStreamOutbound interface {
+		CallStream(ctx context.Context, req *transport.StreamRequest, out UnchainedStreamOutbound) (*transport.ClientStream, error)
+	}
 )
+
+// Outbound is the common interface for all outbounds.
+//
+// Outbounds should also implement the Namer interface so that YARPC can
+// properly update the Request.Transport field.
+type Outbound interface {
+	transport.Lifecycle
+
+	// Transports returns the transports that used by this outbound, so they
+	// can be collected for lifecycle management, typically by a Dispatcher.
+	//
+	// Though most outbounds only use a single transport, composite outbounds
+	// may use multiple transport protocols, particularly for shadowing traffic
+	// across multiple transport protocols during a transport protocol
+	// migration.
+	Transports() []transport.Transport
+}
+
+// UnchainedUnaryOutbound is a transport that knows how to send unary requests for procedure
+// calls.
+type UnchainedUnaryOutbound interface {
```

Comment:
Sorry, we were not explicit enough: the idea was to rename "Unchained" prefix to "Direct" prefix, "UnaryOutbound" may keep the same name though, no real reasons to renamed it to "DirectUnaryOutbound".

I.e.
UnchainedUnaryOutbound -> DirectUnaryOutbound
UnchainedCall -> DirectCall
etc.